extensions/wikia/UserLogin/js/UserSignupAjaxValidation.js

```diff
@@ -0,0 +1,118 @@
+/* global  wgScriptPath, UserSignup */
+(function () {
+	'use strict';
+
+	/**
+	 * UserSignupAjaxValidation is contains business logic for ajax signup validation
+	 * wikiaForm - instance of WikiaForm
+	 * inputsToValide - array of input names to be ok'ed before submission
+	 * submitButton - pointer to main submit button of the form
+	 */
+	var UserSignupAjaxValidation = function (options) {
+		this.wikiaForm = options.wikiaForm;
+		this.inputsToValidate = options.inputsToValidate || [];
+		this.notEmptyFields = options.notEmptyFields || [];
+		this.captchaField = options.captchaField || '';
+		this.submitButton = $(options.submitButton);
+
+		this.activateSubmit();
+	};
+
+	UserSignupAjaxValidation.prototype.validateInput = function (e) {
+		var el = $(e.target),
+			paramName = el.attr('name'),
+			params = this.getDefaultParamsForAjax(),
+			proxyObj;
+
+		params.field = paramName;
+		params[paramName] = el.val();
+
+		proxyObj = {
+			'paramName': paramName,
+			'form': this
+		};
+
+		$.get(wgScriptPath + '/wikia.php', params, $.proxy(this.validationHandler, proxyObj));
```

Comment:
@jsutterfield, I think we should be using `bind()` going forward. Feel free to create a PR in the guidelines repo, or if you prefer, I can do it.